### PR TITLE
NumbersCount.java - Roobini

### PR DESCRIPTION
**Pull request by sroobini**
0 Occurs 1 times
1 Occurs 6 times
2 Occurs 6 times
3 Occurs 3 times
4 Occurs 3 times
5 Occurs 6 times
6 Occurs 6 times
7 Occurs 0 times
8 Occurs 4 times
9 Occurs 11 times

                  # 
                  # 
                  # 
                  # 
                  # 
  # #     # #     # 
  # #     # #     # 
  # #     # #   # # 
  # # # # # #   # # 
  # # # # # #   # # 
# # # # # # #   # # 
0 1 2 3 4 5 6 7 8 9 

0 Occurs 3 times
1 Occurs 5 times
2 Occurs 6 times
3 Occurs 5 times
4 Occurs 6 times
5 Occurs 4 times
6 Occurs 9 times
7 Occurs 2 times
8 Occurs 5 times
9 Occurs 6 times

            #       
            #       
            #       
    #   #   #     # 
  # # # #   #   # # 
  # # # # # #   # # 
# # # # # # #   # # 
# # # # # # # # # # 
# # # # # # # # # # 
0 1 2 3 4 5 6 7 8 9

### CODE REVIEW

File: NumbersCount.java [hunk 1]
```diff
@@ -65,11 +66,122 @@
...
+    public int[] numberCount(String input){
+    	int index = input.length()-1;
+    	String newInput;
+
+    	if(index<0){
+    		
+    		return count;
+    	}
+    	
+    	if(input.charAt(index)=='0'){
```
Comment (chandramouly) [inline NumbersCount.java:118]: Good understanding of recursion, nicely implemented, base case was handled properly..
Can we replace the 10 if conditions with just one :) do we really need to check 0-9 separately?

File: NumbersCount.java [hunk 2]
```diff
@@ -65,11 +66,122 @@
...
+    	}
+    	
+    	
+    	newInput = input.substring(0, index);
+    	numberCount(newInput);
+    	return count;
+    }
+    
+    public void testBench(){
+    	int[] counter = numberCount("2146859296389521599993229915608941463961565182");
```
Comment (chandramouly) [inline NumbersCount.java:165]: Can we add null and empty strings as well? and execute your program???

Comment (sroobini) [reply NumbersCount.java:165]: i tried empty strings. it says 0-9 occurs 0 times.
i didnt try null though.. will check it

File: NumbersCount.java (hunk 1 above)
Comment (sroobini) [reply NumbersCount.java:118]: will think about the if conditions

File: NumbersCount.java (hunk 2 above)
Comment (balamuralim) [reply NumbersCount.java:165]: @sroobini : Try the approach that Mouly suggested.

**Pull request closed without merging**